openid-4-verifiable-presentations-1_0.md

```diff
@@ -432,22 +447,21 @@ Note: "https://self-issued.me/v2" is a symbolic string and can be used as an `au
 
 ## Verifier Metadata Management {#client_metadata_management}
 
-The `client_id_scheme` enables deployments of this specification to use different mechanisms to obtain and validate metadata of the Verifier beyond the scope of [@!RFC6749]. The term `client_id_scheme` is used since the Verifier is acting as an OAuth 2.0 Client.
+The Client Identifier Scheme enables deployments of this specification to use different mechanisms to obtain and validate metadata of the Verifier beyond the scope of [@!RFC6749]. The term Client Identifier Scheme is used since the Verifier is acting as an OAuth 2.0 Client.
 
-This specification defines the following values for the `client_id_scheme` parameter, followed by the examples where applicable: 
+This specification defines the following Client Identifier Schemes, followed by the examples where applicable: 
 
-* `pre-registered`: This value represents the [@!RFC6749] default behavior, i.e., the Client Identifier needs to be known to the Wallet in advance of the Authorization Request. The Verifier metadata is obtained using [@!RFC7591] or through out-of-band mechanisms.
+* `pre-registered`: This value represents the [@!RFC6749] default behavior, i.e., the Client Identifier needs to be known to the Wallet in advance of the Authorization Request. The Verifier metadata is obtained using [@!RFC7591] or through out-of-band mechanisms. Example Client Identifier: `pre-registered:e1fada73-8c43-4745-9756-303c2dc938a6`.
 
-* `redirect_uri`: This value indicates that the Verifier's Redirect URI (or Response URI when Response Mode `direct_post` is used) is also the value of the Client Identifier. The Authorization Request MUST NOT be signed. The Verifier MAY omit the `redirect_uri` Authorization Request parameter (or `response_uri` when Response Mode `direct_post` is used). All Verifier metadata parameters MUST be passed using the `client_metadata` parameter defined in (#vp_token_request).
+* `redirect_uri`: This value indicates that the Client Identifier (without the prefix `redirect_uri:`) is the Verifier's Redirect URI (or Response URI when Response Mode `direct_post` is used). The Authorization Request MUST NOT be signed. The Verifier MAY omit the `redirect_uri` Authorization Request parameter (or `response_uri` when Response Mode `direct_post` is used). All Verifier metadata parameters MUST be passed using the `client_metadata` parameter defined in (#vp_token_request). Example Client Identifier: `redirect_uri:https%3A%2F%2Fclient.example.org%2Fcb`.
```

Comment:
I dont think we should be continuing to support this client ID scheme because its security properties are quite un-usual. Namely because the authorisation request can't be signed and we can't have pre-registered metadata nor a trusted way to resolve the metadata (e.g via a URL) the client is entirely open to impersonation with regard to what it supports from the protocol.

Reply:
I think we could recommend against using it in production (if we don't already), but given it gets quite wide usage in non-production implementations I'm not sure it would be helpful to remove it unless we're replacing it with something that has a similar level of ease of use. (I definitely don't think we should remove it in this PR, it would be out of scope here.)

Reply:
Given the original context in which this issue was raised was due to security concerns with client_id_scheme I do think while we are revisiting this functionality in the specification we should also consider removing it. Personally I've not seen wide usage of this particular client id scheme.

Reply:
It's supported in the conformance suite and about half the tests run have used the redirect_uri client_id_scheme. 

The lower security associated with redirect_uri now only occur when the verifier deliberately uses redirect_uri (rather than their being the potential for redirect_uri to be undetectably used when the verifier hadn't intended it), so we've made progress. Happy to see if the WG would rather remove it, but we should keep that as a separate issue & PR.

Reply:
@tplooker Please file a separate issue for this.

Reply:
Agreed have opened as #269